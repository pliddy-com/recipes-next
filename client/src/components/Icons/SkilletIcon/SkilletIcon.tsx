import Box from '@mui/material/Box';
import SvgIcon from '@mui/material/SvgIcon';

import colors from 'theme/values/colors';

const SkilletIcon = () => (
  <Box
    role="graphics-symbol"
    data-testid="skillet"
    aria-label="skillet"
    className="icon"
  >
    <SvgIcon className="icon">
      <path
        d="M1.818.002c-.272-.018-.572.069-.867.36L.699.607l-.25.248c-.946.93.209 1.916.209 1.916l3.77 3.176c1.1.982 3.103 3.201 4.167 4.4a8.596 8.541 0 1 0 1.819-1.787C9.218 7.49 7.006 5.477 6.03 4.372L2.878.59S2.415.042 1.818.002zM1.741 1.23a.448.445 0 0 1 .317.133l.98.985a.449.446 0 1 1-.638.627l-.98-.985a.45.447 0 0 1 .32-.76Z"
        style={{ fill: colors.primary.main, stroke: 'none' }}
      />
    </SvgIcon>
  </Box>
);

export default SkilletIcon;
