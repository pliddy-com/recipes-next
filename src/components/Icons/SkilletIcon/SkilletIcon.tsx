import { FC } from 'react';

import Box from '@mui/material/Box';
import SvgIcon, { SvgIconProps } from '@mui/material/SvgIcon';

import colors from 'theme/colors';

const SkilletIcon: FC<SvgIconProps> = () => (
  <Box role="graphics-symbol" data-testid="skillet" aria-label="skillet">
    <SvgIcon>
      <path
        d="M10.347 8.561c-1.187-1.07-3.385-3.086-4.354-4.19L2.859.59S1.886-.57.945.362L.696.608l-.25.247c-.94.93.207 1.916.207 1.916L4.4 5.946c1.094.982 3.084 3.202 4.141 4.401a8.54 8.54 0 1 0 1.807-1.786ZM2.185 2.75l-.974-.985a.446.446 0 1 1 .634-.626l.974.984a.446.446 0 1 1-.635.627z"
        style={{ color: colors.primary.main }}
      />
    </SvgIcon>
  </Box>
);

export default SkilletIcon;
