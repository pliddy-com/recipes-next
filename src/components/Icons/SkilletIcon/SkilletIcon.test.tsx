import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import SkilletIcon from './SkilletIcon';

describe('LogoIcon', () => {
  it('loads and displays greeting', () => {
    // Render a React element into the DOM
    const { getByRole, asFragment } = render(<SkilletIcon />);

    // assert that an SVG with an aria role of 'graphics-symbol' exists
    const component = getByRole('graphics-symbol', { name: 'skillet' });
    expect(component).toBeInTheDocument();

    // assert that the component matches the existing snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
