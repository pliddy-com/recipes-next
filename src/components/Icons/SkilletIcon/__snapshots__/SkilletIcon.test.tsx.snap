// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoIcon loads and displays greeting 1`] = `
<DocumentFragment>
  <div
    aria-label="skillet"
    class="MuiBox-root css-0"
    data-testid="skillet"
    role="graphics-symbol"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10.347 8.561c-1.187-1.07-3.385-3.086-4.354-4.19L2.859.59S1.886-.57.945.362L.696.608l-.25.247c-.94.93.207 1.916.207 1.916L4.4 5.946c1.094.982 3.084 3.202 4.141 4.401a8.54 8.54 0 1 0 1.807-1.786ZM2.185 2.75l-.974-.985a.446.446 0 1 1 .634-.626l.974.984a.446.446 0 1 1-.635.627z"
        style="color: rgb(25, 25, 25);"
      />
    </svg>
  </div>
</DocumentFragment>
`;
